modules/github-bots/sdk/github.go

```diff
@@ -115,21 +122,23 @@ func checkRateLimiting(_ context.Context, githubErr error) (bool, time.Duration)
 // API calls. It will handle checking for rate limiting as well as any errors
 // returned by the API.
 func handleGithubResponse(ctx context.Context, resp *github.Response, err error) error {
-	githubErr := github.CheckResponse(resp.Response)
-	if githubErr != nil {
-		rateLimited, delay := checkRateLimiting(ctx, githubErr)
-		// if we were not rate limited, return err
-		if !rateLimited {
-			return err
+	// resp may be nil if err is nonempty. However, err may contain a rate limit
+	// error so we have to inspect for rate limiting if resp is non-nil
+	if resp != nil {
```

Comment:
Does it make sense to have an early return if `resp == nil`, instead of enclosing the whole block in an if statement?

Reply:
I beg to differ on that, but this is a code style nit too